Octokit/Http/Response.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Net;
+
+namespace Octokit.Internal
+{
+    /// <summary>
+    /// Represents a generic HTTP response
+    /// </summary>
+    public class Response : IResponse
```

Comment:
Ideally I'd like everything under `Octokit.Internal` to be marked as internal - my thinking on this is that if there's a compelling case to make a type public, it shouldn't be in this namespace.

Happy to do a once-over on this after this PR (before the next major release), just calling this out as it's a new type.